### PR TITLE
Quick Filter Fixes:

### PR DESCRIPTION
**Pull request by aydreeihn**
Make the following quick filters work:
- Assigned
- Assignee
- Department Manager
- Assigned Team

Also, make sure we account for if there are no choices available for a quick filter 
Ex: The helpdesk does not have any teams

### CODE REVIEW

File: include/class.search.php [hunk 1]
```diff
@@ -1399,8 +1406,14 @@ class DepartmentManagerSelectionField extends AgentSelectionField {
     static $_members;
 
     function getChoices($verbose=false) {
-        if (isset($this->_members))
-            $this->_members = Staff::getStaffMembers();
+        if (!isset($this->_members)) {
+            $managers = array();
+            foreach (Staff::objects() as $staff) {
```
Comment (protich) [inline include/class.search.php]: Why not filter via ORM and select only managers?

File: include/class.forms.php [hunk 2]
```diff
@@ -1980,6 +1980,24 @@ function getQuickFilterChoices() {
...
 
     function applyQuickFilter($query, $qf_value, $name=false) {
+        global $thisstaff;
+
+        //special assignment quick filters
+        switch (true) {
+            case ($qf_value == 'assigned'):
+            case ($qf_value == '!assigned'):
+                $result = AssigneeChoiceField::getSearchQ($qf_value, $qf_value);
+                return $query->filter($result);
```
Comment (greezybacon) [inline include/class.forms.php]: Shouldn't this be defined in `AssigneeChoiceField::applyQuickFilter`?

Comment (aydreeihn) [reply include/class.forms.php]: We could do that, but we would end up needing to define an applyQuickFilter function for AssignedField, AssigneeChoiceField, and TeamSelectionField. 

applyQuickFilter is hit in class.queue.php getQuery 
$qf->applyQuickFilter()
where $qf could be one of those classes depending on the quick filter.

After discussing it, it looks like it'll be fine to just leave it as is.

File: include/class.search.php (hunk 1 above)
Comment (aydreeihn) [reply include/class.search.php]: Off of departments?

```
if (!isset($this->_members)) {
            $managers = array();
            $managerIds = Dept::objects()
                ->filter(Q::any(array(
                    'manager_id__gt' => 0,
                )))
                ->values_flat('manager_id');

            foreach ($managerIds as $key => $id) {
                $staff = Staff::lookup($id[0]);
                $managers['s'.$id[0]] = $staff->getName()->name;
            }
            $this->_members = $managers;
        }
```

Does this make it less expensive? I still ended up with a lookup on the staff member.

**Pull request merged**